src/ast/expression.rs

```diff
@@ -3,25 +3,31 @@ use crate::{
   token::{Grammar, Keyword, Operator, TokenStream},
 };
 
-use super::{identifier::Identifier, AstError, AstResult, Location};
+use super::{
+  identifier::{self, Identifier},
+  AstError, AstResult, Location,
+};
 
 #[derive(Debug, PartialEq, Clone)]
 pub enum Expression<'a> {
   Bool(bool, Location),
   String(&'a str, Location), // TODO: given we want to reverse, maybe use owned?
   Number(f64, Location),
+  List(Vec<Expression<'a>>, Location),
+  Map(Vec<(Identifier, Expression<'a>)>, Location),
+  Index(Identifier, Vec<Expression<'a>>, Location),
```

Comment:
Probably expand this in to a struct variant, rather than tuple (i.e. use named fields). Is `Identifier` sufficient, or is it possible to index an expression?